src/figuresClasses.ts

```diff
@@ -1,19 +1,114 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Triangle = 'triangle',
+}
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+abstract class BaseFigure implements Figure {
+  abstract readonly shape: Shape;
+
+  abstract color: Color;
 
+  abstract getArea(): number;
+
+  protected checkAllSides(sides: number[]): void {
+    if (sides.some((side) => side <= 0)) {
+      throw new Error(
+        `Figure of Shape ${this.shape} should have only positive params`,
+      );
+    }
+  }
 }
 
-export class Circle {
+export class Triangle extends BaseFigure {
+  readonly shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    super();
+
+    this.checkAllSides([a, b, c]);
+
+    const [
+      sideA,
+      sideB,
+      sideC,
+    ]: number[] = [a, b, c].sort((prev, curr) => prev - curr);
+
+    if (sideC >= sideB + sideA) {
+      throw new Error('Invalid sides length!');
+    }
+  }
 
+  getArea(): number {
+    const semiPerimetr = (this.a + this.b + this.c) / 2;
+
+    const area = Math.sqrt(
+      semiPerimetr
+      * (semiPerimetr - this.a)
+      * (semiPerimetr - this.b)
+      * (semiPerimetr - this.c),
+    );
+
+    return +area.toFixed(2);
+  }
 }
 
-export class Rectangle {
+export class Circle extends BaseFigure {
+  readonly shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    super();
+
+    this.checkAllSides([radius]);
+  }
+
+  getArea(): number {
+    const area = Math.PI * (this.radius ** 2);
 
+    return Math.floor(area * 100) / 100;
+  }
```

Comment:
maybe you need a separate function for this... but that's only my opinion=))

Reply:
agree